### PR TITLE
Change WebRequest example with HttpClient

### LINKED ISSUE #27871: Code example deprecated
[WebRequest is deprecated, is it possible to make an example using HttpClient?]

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 4c03e434-07bb-8309-7381-ba81ae82de73
* Version Independent ID: a0f35eb7-7253-1025-dd90-59d982654ba1
* Content: [NTLM and Kerberos Authentication - .NET Framework](https://docs.microsoft.com/en-us/dotnet/framework/network-programming/ntlm-and-kerberos-authentication)
* Content Source: [docs/framework/network-programming/ntlm-and-kerberos-authentication.md](https://github.com/dotnet/docs/blob/main/docs/framework/network-programming/ntlm-and-kerberos-authentication.md)
* Product: **dotnet-framework**
* Technology: **dotnet-networking**
* GitHub Login: @karelz
* Microsoft Alias: **ncldev**

### PR DESCRIPTION
**Pull request by liveans**
## Summary

Changed deprecated WebRequest class with HttpClient in example.

Fixes #27871.

### CODE REVIEW

File: docs/framework/network-programming/ntlm-and-kerberos-authentication.md [hunk 1]
```diff
@@ -26,31 +26,57 @@ ms.assetid: 9ef65560-f596-4469-bcce-f4d5407b55cd
...
+string myUri = "http://www.contoso.com/";
+using (HttpClientHandler handler = new HttpClientHandler()
+{
+    Credentials = new NetworkCredential(UserName, SecurelyStoredPassword, Domain),
+})
+using (HttpClient client = new HttpClient(handler))
+{
+    string result = client.GetStringAsync(myUri).Result;
+    // Do Other Stuff...
+}
```
Comment (IEvangelist) [inline docs/framework/network-programming/ntlm-and-kerberos-authentication.md:51]:
```suggestion
string myUri = "http://www.contoso.com/";
using HttpClientHandler handler = new()
{
    Credentials = new NetworkCredential(UserName, SecurelyStoredPassword, Domain),
};
using HttpClient client = new(handler);
string result = await client.GetStringAsync(myUri);
// Do Other Stuff...
```

File: docs/framework/network-programming/ntlm-and-kerberos-authentication.md [hunk 2]
```diff
@@ -26,31 +26,57 @@ ms.assetid: 9ef65560-f596-4469-bcce-f4d5407b55cd
...
+string myUri = "http://www.contoso.com/";
+using (HttpClientHandler handler = new HttpClientHandler()
+{
+    Credentials = CredentialCache.DefaultCredentials,
+})
+using (HttpClient client = new HttpClient(handler))
+{
+    string result = client.GetStringAsync(myUri).Result;
+    // Do Other Stuff...
+}
```
Comment (IEvangelist) [inline docs/framework/network-programming/ntlm-and-kerberos-authentication.md:79]:
```suggestion
string myUri = "http://www.contoso.com/";
using HttpClientHandler handler = new()
{
    Credentials = CredentialCache.DefaultCredentials,
};
using HttpClient client = new(handler);
string result = await client.GetStringAsync(myUri);
// Do Other Stuff...
```

File: docs/framework/network-programming/ntlm-and-kerberos-authentication.md [hunk 3]
```diff
@@ -26,31 +26,57 @@ ms.assetid: 9ef65560-f596-4469-bcce-f4d5407b55cd
...
-Dim WReq As WebRequest = WebRequest.Create(MyURI)  
-WReq.Credentials = _  
-    New NetworkCredential(UserName, SecurelyStoredPassword, Domain)  
+Dim myUri As String = "http://www.contoso.com/"  
+Using handler As New HttpClientHandler()
+    With handler
+        .Credentials = New NetworkCredential(UserName, SecurelyStoredPassword, Domain)
+    End With
+    Using client As New HttpClient(handler)
+        Dim result As String = client.GetStringAsync(myUri).Result
```
Comment (IEvangelist) [inline docs/framework/network-programming/ntlm-and-kerberos-authentication.md:35]:
```suggestion
        Dim result As String = client.GetStringAsync(myUri).GetAwaiter().GetResult()
```

File: docs/framework/network-programming/ntlm-and-kerberos-authentication.md [hunk 4]
```diff
@@ -26,31 +26,57 @@ ms.assetid: 9ef65560-f596-4469-bcce-f4d5407b55cd
...
-Dim MyURI As String = "http://www.contoso.com/"  
-Dim WReq As WebRequest = WebRequest.Create(MyURI)  
-WReq.Credentials = CredentialCache.DefaultCredentials  
+Dim myUri As String = "http://www.contoso.com/"  
+Using handler As New HttpClientHandler()
+    With handler
+        .Credentials = CredentialCache.DefaultCredentials
+    End With
+    Using client As New HttpClient(handler)
+        Dim result As String = client.GetStringAsync(myUri).Result
```
Comment (IEvangelist) [inline docs/framework/network-programming/ntlm-and-kerberos-authentication.md:63]:
```suggestion
        Dim result As String = client.GetStringAsync(myUri).GetAwaiter().GetResult()
```

**Review by IEvangelist — CHANGES_REQUESTED** [review]
This looks good, thank you. We'll :shipit: when you make the suggested changes.

File: docs/framework/network-programming/ntlm-and-kerberos-authentication.md [hunk 5]
```diff
@@ -26,31 +26,53 @@ ms.assetid: 9ef65560-f596-4469-bcce-f4d5407b55cd
...
-Dim WReq As WebRequest = WebRequest.Create(MyURI)  
-WReq.Credentials = _  
-    New NetworkCredential(UserName, SecurelyStoredPassword, Domain)  
+Dim myUri As String = "http://www.contoso.com/"  
+Using handler As New HttpClientHandler()
+    With handler
+        .Credentials = New NetworkCredential(UserName, SecurelyStoredPassword, Domain)
+    End With
+    Using client As New HttpClient(handler)
+        Dim result As String = client.GetStringAsync(myUri).GetAwaiter().GetResult()
```
Comment (MihaZupan) [inline docs/framework/network-programming/ntlm-and-kerberos-authentication.md:35]: I don't know anything about VB, but based on [these docs](https://learn.microsoft.com/en-us/dotnet/visual-basic/programming-guide/concepts/async/#BKMK_HowtoWriteanAsyncMethod) it seems like we should be able to use Await here as well (so we don't accidentally promote "sync over async" patterns).

Comment (IEvangelist) [reply docs/framework/network-programming/ntlm-and-kerberos-authentication.md:35]: I agree if that is available. I didn't know for sure.

Comment (IEvangelist) [inline docs/framework/network-programming/ntlm-and-kerberos-authentication.md:35]:
```suggestion
        Dim result As String = Await client.GetStringAsync(myUri)
```

File: docs/framework/network-programming/ntlm-and-kerberos-authentication.md [hunk 6]
```diff
@@ -26,31 +26,53 @@ ms.assetid: 9ef65560-f596-4469-bcce-f4d5407b55cd
...
-Dim MyURI As String = "http://www.contoso.com/"  
-Dim WReq As WebRequest = WebRequest.Create(MyURI)  
-WReq.Credentials = CredentialCache.DefaultCredentials  
+Dim myUri As String = "http://www.contoso.com/"  
+Using handler As New HttpClientHandler()
+    With handler
+        .Credentials = CredentialCache.DefaultCredentials
+    End With
+    Using client As New HttpClient(handler)
+        Dim result As String = client.GetStringAsync(myUri).GetAwaiter().GetResult()
```
Comment (IEvangelist) [inline docs/framework/network-programming/ntlm-and-kerberos-authentication.md:61]:
```suggestion
        Dim result As String = Await client.GetStringAsync(myUri)
```

**Review by IEvangelist — APPROVED** [review]
A few more nits.

**Pull request merged**